linkerd/proxy/discover/src/buffer.rs

```diff
@@ -61,12 +59,11 @@ where
         self.inner.poll_ready()
     }
 
+    #[inline]
```

Comment:
It seems like `#[inline]`s are starting to show up more and we don't really have any benchmarks to confirm it's an optimization. Is there any reason this seemed like a necessary addition? I'm not recommending to remove it if not, but just want to make sure I didn't miss an earlier discussion.

Reply:
@hawkw tells me to add them so i add them. i have no clue what i'm doing.

Reply:
FWIW, explicit `#[inline]`s are _generally_ only necessary on public fns that are called across crate boundaries. For private functions, I think we can usually trust rustc to do the right thing, except in edge cases.

Reply:
Okay, per discussion on slack, I'll probably omit: our default should be to omit unless we've measured an impact.